Waitscreen while loading

kris10ansn opened the pull request:
This is my first time doing a pull request, ever 😅 

This commit makes the wait screen show up while loading. When the level is done loading the screen stays for however many seconds there are left. If the loading takes more than two seconds it should disappear instantly.

Now there are some caveats:
- I modified the `createPlayerProgressLayer `function to rather take player and levelname than the whole level as an argument (the level isn't available before loading). 
- I created a `createDummyDashboardLayer `for the waitscreen as the `createDashboardLayer `needs access to the level to get ahold of the leveltimer. This could be done in other ways too, like adding the dashboard after the level is loaded. This would not be as seemless unless you are hosting the game on your own computer with little to no loading time.. There are probably better ways to do this, but i simply didn't have time to explore it.

pomle committed:
Sync with published result of episode 24.

pomle commented:
Very cool! It looks like this PR is missing your changes. Let me know in the [char](https://discord.gg/aeh8DTv) if you need help.

pomle committed:
Refactor entity loader

pomle committed:
Remove unused contructor

pomle committed:
Use async/await for loadEntities

pomle committed:
Simplify setup function in loadEntities

bouveng committed:
New tiles.png. Updated spritesheet and patterns.

bouveng committed:
Added new sprites.png. Updated spritesheets.

bouveng committed:
Added new music and sound effects.

bouveng committed:
Added font for points.

bouveng committed:
Added new font. Added compability for text output.

bouveng committed:
Fixed positioning of text in dashboard.

bouveng committed:
Fixed positioning of x and mario in progresslayer.

bouveng committed:
Made coin use new traits getter.

bouveng committed:
Fixed total time and speed for time counter.

bouveng committed:
Correct jump height for Mario.

bouveng committed:
Spawn entities when they enter the screen.

bouveng committed:
Goomba blue, brown. Koopa green, blue.

bouveng committed:
Changed flip for bullet to work with new sprite.

bouveng committed:
Tile definition "empty" renamed to "metal".

bouveng committed:
Add Levels, Patterns, and Music sheets

pomle committed:
Add SpriteSheet for Brick Shrapnel

pomle committed:
Add initial BrickShrapnel entity

pomle committed:
Load BrickShrapnel

pomle committed:
Update brick tile to emit shrapnel

pomle committed:
Show Bricks staying in the level indefinitely

pomle committed:
Add LifeLimit trait

pomle committed:
Add LifeLimit trait to BrickShrapnel

pomle committed:
Add BrickShrapnel SoundBoard

pomle committed:
Add Brick Destroy sound to Brick tile

pomle committed:
Remove position logging

pomle committed:
Reintroduce entity loader promise chain

Needed for composability with pool

pomle committed:
Add entity pooling and set size to 2

pomle committed:
Increase brick pool to 2

bouveng committed:
Force uppercase in font class. Using × instead of x.

pomle committed:
Add Keymap to input setup

Increases readability.

pomle committed:
Only call draw if exists

pomle committed:
Add equals method to Vec2

pomle committed:
Add direction enum

pomle committed:
Add PipePortal SoundBoard

pomle committed:
Add Pipe and PipeTraveller trait

pomle committed:
Add PipeTraveller trait to Mario

pomle committed:
Route input to PipeTraveller trait

pomle committed:
Add PipePortal entity

pomle committed:
Add PipePortal entity to Pipe debug level

pomle committed:
Set brick pool to 8

pomle committed:
Keep track of travellers in pipe

pomle committed:
Remove PlayerEnv/Controller

pomle committed:
Never remove Mario when killed

pomle committed:
Add Travelling animation to Pipe

pomle committed:
No function parameter overwrite

bouveng committed:
Use pattern chocolate-stack-4h.

bouveng committed:
Remove space from x in player-progress.

pomle committed:
Remove duplicate EventBuffer from constructor

pomle committed:
Use direction from props

pomle committed:
Throw error when entity not found

pomle committed:
Add Event to Pipe

pomle committed:
Add Event complete to Level

bouveng committed:
Added missing small coin icon for underworld.

pomle committed:
Copy props to Entity for pipe

pomle committed:
Connect coin-room-1 pipe

pomle committed:
Remove unused imports

pomle committed:
Clamp camera scroll to level bounds

pomle committed:
Avoid repeating traits.get(Go) in Mario anim

pomle committed:
Handle Mario animation with Pipe

pomle committed:
Add test pipes

bouveng committed:
Changed playerPosition to checkpoints.

pomle committed:
Handle checkpoints

bouveng committed:
Checkpoints is array of arrays.

pomle committed:
Move LevelTimer to Player

pomle committed:
Player world

pomle committed:
Paint sprites one layer below last layer

bouveng committed:
Correct position for "WORLD" and level name.

pomle committed:
Provide default checkpoint

pomle committed:
Add tick to GameContext

pomle committed:
Keep track if LevelTimer was used on level

pomle committed:
Provide auto enter pipe setting

pomle committed:
Provide explicit enter pipe method

bouveng committed:
Add missing princess.ogg.

pomle committed:
Replace auto enter with linking explicitly to pipe

pomle committed:
Support getting and setting center of BoundingBox

Probably should consider offset now, but I can't figure
out the math at this time. :)

pomle committed:
Add Align-utility

pomle committed:
Increase PipePortal height

pomle committed:
Add connectEntity utility to Pipe trait

pomle committed:
Only allow entering pipe from within bounds

pomle committed:
Route Mario anim properly so as to not jump while travelling

pomle committed:
Tweak debug level

pomle committed:
Add FlagPole SoundBoard

pomle committed:
Add FlagPole entity

pomle committed:
Load flag pole

pomle committed:
Add flag debug level

pomle committed:
Add meridian and equator to BoundingBox

pomle committed:
Add Pole trait to Pole entity

pomle committed:
Add PoleTraveller to Mario

pomle committed:
Handle heading for pole

pomle committed:
Show climb frame when pole sliding

pomle committed:
Add Pole and PoleTraveller traits

bouveng committed:
Add fixed aspect ratios for screen.

bouveng committed:
Move css from index to main.css

bouveng committed:
Only play pipe.ogg when entering pipes.

bouveng committed:
Rename tile "type" to "behavior".

pomle committed:
Add Piranha Plant sprite sheet

pomle committed:
Add Piranha Plant logic

pomle committed:
Add distance method to Vec2

pomle committed:
Load Piranha Plant

pomle committed:
Add Piranha Plant to level

pomle committed:
Include last tile's width when calculating camera pan max

pomle committed:
Animate climb when on flag pole

bouveng committed:
Rename tile "name" to "style".

bouveng committed:
Fix typo in underwater-pattern.

bouveng committed:
Add cheep-cheep sprite sheets.

bouveng committed:
Add cheep-cheep entity.

bouveng committed:
Load cheep slow, fast, slow-wavy and fast-wavy.

bouveng committed:
Add cheep-cheep to level.

bouveng committed:
Fix coin range position.

bouveng committed:
Add empty tile layer.

bouveng committed:
Add linewidth 1 + no sub pixel rendering.

pomle committed:
Create gcs-deploy.yml (#134)

pomle committed:
Update gcs-deploy.yml

pomle committed:
Start at level 1-1

pomle committed:
Set cache control on uploads

pomle committed:
Set wait time
